test/create-identity-response-handler-test.ts

```diff
@@ -0,0 +1,126 @@
+import * as assert from 'assert'
+import { createIdentityResponseHandler, IdentityResponseScenarios } from '../lib/create-identity-response-handler'
+import { Scenario } from '../lib/verify-service-provider-api/translated-response-body'
+import * as td from 'testdouble'
+
+function verifyNotCalled (fn: any) {
+  td.verify(fn(), { times: 0, ignoreExtraArgs: true })
+}
+
```

Comment:
Repeating a review comment which got lost in a force-push:

The two different versions of create[Identity]ResponseHandler work on values from the Scenario enum, which is shared by both journey flows (matching and non-matching).  Some values in that enum are only applicable to one of the journey flows.  We don't have any handling which determines what should happen if one of the functions encounters a scenario value from the wrong journey flow, and we also have no tests for that.

Is that something we're comfortable with?  I think a similar issue exists in other areas of passport-verify, but it's more obvious here as the test coverage in this area is otherwise very comprehensive.

Reply:
I think it's a good thing to have, but we can add the restriction as part of a follow-up story.